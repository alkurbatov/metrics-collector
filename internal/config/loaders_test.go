package config_test

import (
	"testing"

	"github.com/alkurbatov/metrics-collector/internal/config"
	"github.com/alkurbatov/metrics-collector/internal/entity"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestLoadConfigFromFile(t *testing.T) {
	tt := []struct {
		name string
		path entity.FilePath
		cfg  config.Config
	}{
		{
			name: "Load agent config",
			path: "../../configs/agent.example.json",
			cfg:  &config.Agent{},
		},
		{
			name: "Load server config",
			path: "../../configs/server.example.json",
			cfg:  &config.Server{},
		},
	}

	for _, tc := range tt {
		t.Run(tc.name, func(t *testing.T) {
			err := config.LoadFromFile(tc.path, tc.cfg)

			require.NoError(t, err)
			snaps.MatchSnapshot(t, tc.cfg.String())
		})
	}
}

func TestLoadUnexistingConfig(t *testing.T) {
	cfg := &config.Agent{}

	err := config.LoadFromFile("xxx", cfg)
	require.Error(t, err)
}
