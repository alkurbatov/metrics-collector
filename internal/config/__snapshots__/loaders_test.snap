
[TestLoadConfigFromFile/Load_agent_config - 1]
Configuration:
        Poll interval: 1s
        Report interval: 1s
        Collector address: localhost:8080
        Public key path: ./build/keys/public.pem
        PollTimeout: 0.000000s
        ExportTimeout: 0.000000s
        Debug: true
---

[TestLoadConfigFromFile/Load_server_config - 1]
Configuration:
        HTTP API address: localhost:8080
        gRPC API address: localhost:8080
        Store interval: 1s
        Store path: /path/to/file.db
        Restore on start: true
        Private key path: ./build/keys/private.pem
        Trusted subnet: 192.168.0.0/16
        Debug: true

---
